### PR TITLE
fix(components): [table-v2] dynamic height causes overall calculation errors

### LINKED ISSUE #18655: [Component] [table-v2]  冻结列在动态行高时，列高度会小于内容，导致有的行没有显示
Bug Type: **`Component`**

## Environment

- Vue Version: `3.5.12`
- Element Plus Version: `2.8.6`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-table-v2`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGUtdjJcbiAgICA6Y29sdW1ucz1cImNvbHVtbnNcIlxuICAgIDpkYXRhPVwiZGF0YVwiXG4gICAgOnNvcnQtYnk9XCJzb3J0XCJcbiAgICA6ZXN0aW1hdGVkLXJvdy1oZWlnaHQ9XCI0MFwiXG4gICAgOndpZHRoPVwiNzAwXCJcbiAgICA6aGVpZ2h0PVwiNDAwXCJcbiAgICBmaXhlZFxuICAgIEBjb2x1bW4tc29ydD1cIm9uQ29sdW1uU29ydFwiXG4gIC8+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c3hcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7XG4gIEVsQnV0dG9uLFxuICBFbFRhZyxcbiAgVGFibGVWMkZpeGVkRGlyLFxuICBUYWJsZVYyU29ydE9yZGVyLFxufSBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmltcG9ydCB0eXBlIHsgQ29sdW1uLCBTb3J0QnkgfSBmcm9tICdAZWxlbWVudC1wbHVzL2NvbXBvbmVudHMvdGFibGUtdjInXG5cbmNvbnN0IGxvbmdUZXh0ID1cbiAgJ1F1YWVyYXQgaXBzYW0gbmVjZXNzaXRhdGlidXMgZXVtIHF1aWJ1c2RhbSBlc3QgaWQgdm9sdXB0YXRlbSBjdW1xdWUgbW9sbGl0aWEuJ1xuY29uc3QgbWlkVGV4dCA9ICdDb3JydXB0aSBkb2xvcmVtcXVlIGEgcXVvcyB2ZXJvIGRlbGVjdHVzIGNvbnNlcXVhdHVyLidcbmNvbnN0IHNob3J0VGV4dCA9ICdFaXVzIG9wdGlvIGZ1Z2lhdC4nXG5cbmNvbnN0IHRleHRMaXN0ID0gW3Nob3J0VGV4dCwgbWlkVGV4dCwgbG9uZ1RleHRdXG5cbi8vIGdlbmVyYXRlIHJhbmRvbSBudW1iZXIgaW4gcmFuZ2UgMCB0byAyXG5cbmxldCBpZCA9IDBcblxuY29uc3QgZGF0YUdlbmVyYXRvciA9ICgpID0+ICh7XG4gIGlkOiBgcmFuZG9tLSR7KytpZH1gLFxuICBuYW1lOiAnVG9tJyxcbiAgZGF0ZTogJzIwMTYtMDUtMDMnLFxuICBkZXNjcmlwdGlvbjogdGV4dExpc3RbTWF0aC5mbG9vcihNYXRoLnJhbmRvbSgpICogMyldLFxufSlcblxuY29uc3QgY29sdW1uczogQ29sdW1uPGFueT5bXSA9IFtcbiAge1xuICAgIGtleTogJ2lkJyxcbiAgICB0aXRsZTogJ0lkJyxcbiAgICBkYXRhS2V5OiAnaWQnLFxuICAgIHdpZHRoOiAxNTAsXG4gICAgc29ydGFibGU6IHRydWUsXG4gICAgZml4ZWQ6IFRhYmxlVjJGaXhlZERpci5MRUZULFxuICB9LFxuICB7XG4gICAga2V5OiAnbmFtZScsXG4gICAgdGl0bGU6ICdOYW1lJyxcbiAgICBkYXRhS2V5OiAnbmFtZScsXG4gICAgd2lkdGg6IDE1MCxcbiAgICBhbGlnbjogJ2NlbnRlcicsXG4gICAgY2VsbFJlbmRlcmVyOiAoeyBjZWxsRGF0YTogbmFtZSB9KSA9PiA8RWxUYWc+e25hbWV9PC9FbFRhZz4sXG4gIH0sXG4gIHtcbiAgICBrZXk6ICdkZXNjcmlwdGlvbicsXG4gICAgdGl0bGU6ICdEZXNjcmlwdGlvbicsXG4gICAgZGF0YUtleTogJ2Rlc2NyaXB0aW9uJyxcbiAgICB3aWR0aDogMTUwLFxuICAgIGNlbGxSZW5kZXJlcjogKHsgY2VsbERhdGE6IGRlc2NyaXB0aW9uIH0pID0+IChcbiAgICAgIDxkaXYgc3R5bGU9XCJwYWRkaW5nOiAxMHB4IDA7XCI+e2Rlc2NyaXB0aW9ufTwvZGl2PlxuICAgICksXG4gIH0sXG4gIHtcbiAgICBrZXk6ICdvcGVyYXRpb25zJyxcbiAgICB0aXRsZTogJ09wZXJhdGlvbnMnLFxuICAgIGNlbGxSZW5kZXJlcjogKCkgPT4gKFxuICAgICAgPD5cbiAgICAgICAgPEVsQnV0dG9uIHNpemU9XCJzbWFsbFwiPkVkaXQ8L0VsQnV0dG9uPlxuICAgICAgICA8RWxCdXR0b24gc2l6ZT1cInNtYWxsXCIgdHlwZT1cImRhbmdlclwiPlxuICAgICAgICAgIERlbGV0ZVxuICAgICAgICA8L0VsQnV0dG9uPlxuICAgICAgPC8+XG4gICAgKSxcbiAgICB3aWR0aDogMTUwLFxuICAgIGFsaWduOiAnY2VudGVyJyxcbiAgfSxcbl1cbmNvbnN0IGRhdGEgPSByZWYoXG4gIEFycmF5LmZyb20oeyBsZW5ndGg6IDUgfSlcbiAgICAubWFwKGRhdGFHZW5lcmF0b3IpXG4gICAgLnNvcnQoKGEsIGIpID0+IChhLm5hbWUgPiBiLm5hbWUgPyAxIDogLTEpKVxuKVxuXG5jb25zdCBzb3J0ID0gcmVmPFNvcnRCeT4oeyBrZXk6ICduYW1lJywgb3JkZXI6IFRhYmxlVjJTb3J0T3JkZXIuQVNDIH0pXG5cbmNvbnN0IG9uQ29sdW1uU29ydCA9IChzb3J0Qnk6IFNvcnRCeSkgPT4ge1xuICBjb25zdCBvcmRlciA9IHNvcnRCeS5vcmRlciA9PT0gJ2FzYycgPyAxIDogLTFcbiAgY29uc3QgZGF0YUNsb25lID0gWy4uLmRhdGEudmFsdWVdXG4gIGRhdGFDbG9uZS5zb3J0KChhLCBiKSA9PiAoYVtzb3J0Qnkua2V5XSA+IGJbc29ydEJ5LmtleV0gPyBvcmRlciA6IC1vcmRlcikpXG4gIHNvcnQudmFsdWUgPSBzb3J0QnlcbiAgZGF0YS52YWx1ZSA9IGRhdGFDbG9uZVxufVxuPC9zY3JpcHQ+XG4iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0AyLjguNi9kaXN0L2luZGV4LmNzcycsICdodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0AyLjguNi90aGVtZS1jaGFsay9kYXJrL2Nzcy12YXJzLmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAMi44LjYvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0AyLjguNi9cIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

1. 虚拟表格有冻结的列
2. 虚拟表格有动态高度的行
3. 虚拟表的行数总高度未充满容器

### What is Expected?

冻结列的高度跟表格内容一样高

### What is actually happening?

冻结列的高度小于表格内容的高度，冻结列内的行显示不全

### Additional comments

CleanShot 2024-10-24 at 17 10 03@2x

### PR DESCRIPTION
**Pull request by ghost**
虚拟表格同时设置了动态行高和冻结列时，计算fixedTable高度时使用estimatedRowHeight乘以行数，而实际行高与estimatedRowHeight存在差异。修复方案是累加实际行高得到表格的实际rowsHeigt

closed #18655

Please make sure these boxes are checked before submitting your PR, thank you!

- [✅] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [✅ ] Make sure you are merging your commits to `dev` branch.
- [✅ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by hanchao-c0ldwave** [comment]
大佬，这条有空也帮忙审一下呗 @btea

**Comment by btea** [comment]
The code has not yet been modified, and the current modification will cause an error.

**Comment by hanchao-c0ldwave** [comment]
> The code has not yet been modified, and the current modification will cause an error.

I have optimized the code logic by passing rowsHeight as a parameter to useStyles for calculating fixedTableHeight and mainTableHeight, rather than abstracting both fixedTableHeight and mainTableHeight into useTable. Before submitting the code, I ran unit tests and lint checks, and rebuilt the project without encountering any issues. Could you please review it again? Thank you very much.

File: packages/components/table-v2/src/use-table.ts [hunk 1]
```diff
@@ -84,24 +84,37 @@ function useTable(props: TableV2Props) {
     resetAfterIndex,
   })
 
+  const rowsHeight = computed(() => {
+    const { estimatedRowHeight, rowHeight } = props
+    const _data = unref(data)
+    if (isNumber(estimatedRowHeight)) {
```
Comment (btea) [inline packages/components/table-v2/src/use-table.ts:90]: `estimatedRowHeight` seems to be useless below, can it be removed?

Comment (hanchao-c0ldwave) [reply packages/components/table-v2/src/use-table.ts:90]: estimatedRowHeight is used to determine whether the row heights are dynamic. rowHeights only have values when estimatedRowHeight is set to a specific number. When estimatedRowHeight is undefined, it's necessary to use _data.length * rowHeight to calculate the total height of all rows. It appears that estimatedRowHeight cannot be removed.

Comment (btea) [reply packages/components/table-v2/src/use-table.ts:90]: OK, it looks fine to me. But it still needs to be reviewed by others, please be patient.

**Review by btea — APPROVED** [review]
Thanks!

**Review by makedopamine — APPROVED** [review]
LGTM

**Comment by Ziesel777** [comment]
This fix somehow broke all my tables. Returned to 2.9.0

**Pull request merged**